cg/models/downsample/downsample_data.py

```diff
@@ -0,0 +1,161 @@
+"""Model for down sampling meta data."""
+import logging
+from pathlib import Path
+
+from cg.apps.downsample.utils import case_exists_in_statusdb
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.constants import Priority
+from cg.store import Store
+from cg.store.models import ApplicationVersion, Family, Sample
+from cg.utils.calculations import multiply_by_million
+
+LOG = logging.getLogger(__name__)
+
+
+class DownsampleData:
+    def __init__(
+        self,
+        status_db: Store,
+        hk_api: HousekeeperAPI,
+        sample_id: str,
+        number_of_reads: float,
+        case_id: str,
+        out_dir: Path,
+    ):
+        """Initialize the downsample data and perform integrity checks.
+        Raises:
+            ValueError
+            FileExistsError
```

Comment:
I think we should write error raised only in function where the error is originated and not in the functions where the error is propagated, but I am not sure. I am just thinking that if we propagate the error in the docstrings we will end up with functions raising many errors.

Reply:
Good point for discussion! I lean towards wanting some indication that errors can be propagated by this method, since for a user of this method, as little of the implementation as possible should be needed. But I don't know what the general consensus is.